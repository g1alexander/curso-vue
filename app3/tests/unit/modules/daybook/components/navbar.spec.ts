import NavbarVue from "@/modules/daybook/components/Navbar.vue";
import { shallowMount } from "@vue/test-utils";
import { createVuexStore } from "../../../mocks/mock-store";
import { createRouterMock, injectRouterMock } from "vue-router-mock";

describe("pruebas en navbar componente", () => {
  const store = createVuexStore({
    user: {
      name: "test",
      email: "test@tes.com",
    },
    idtoken: "ABAC",
    refreshtoken: "123-aba",
    status: "authenticated",
  });

  // create one mock instance, pass options
  const router = createRouterMock();
  beforeEach(() => {
    // inject it globally to ensure `useRoute()`, `$route`, etc work
    // properly and give you access to test specific functions
    injectRouterMock(router);
  });

  test("debe de mostrar el componente correctamente", () => {
    const wrapper = shallowMount(NavbarVue, {
      global: {
        plugins: [store],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("click en logout, debe de cerrar session y redireccionar", async () => {
    const wrapper = shallowMount(NavbarVue, {
      global: {
        plugins: [store],
      },
    });

    await wrapper.find("button").trigger("click");

    expect(store.state.authModule).toEqual({
      user: null,
      status: "not-authenticated",
      idtoken: null,
      refreshtoken: null,
    });

    expect(wrapper.vm.router.push).toHaveBeenCalledWith({ name: "auth-login" });
  });
});
