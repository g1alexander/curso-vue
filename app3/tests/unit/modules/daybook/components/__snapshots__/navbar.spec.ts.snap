// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en navbar componente debe de mostrar el componente correctamente 1`] = `
<nav class="navbar bg-primary"><a href="navbar-brand"><img alt="logo" class="d-inline-block align-text-top mx-2" height="24"><span class="text-white">test</span></a>
  <div class="d-flex"><button class="btn btn-outline-info mx-2"><i class="fa fa-sign-out-alt"></i></button></div>
</nav>
`;
